theano/tensor/nnet/opt.py

```diff
@@ -254,315 +251,6 @@ def local_conv2d_cpu(node):
     return [rval]
 
 
-@local_optimizer([AbstractConv3d])
-def local_conv3d_cpu(node):
-    if not isinstance(node.op, AbstractConv3d):
-        return None
-
-    img, kern = node.inputs
-    if ((not isinstance(img.type, TensorType) or
-         not isinstance(kern.type, TensorType))):
-        return None
-    if node.op.border_mode not in ['valid', (0, 0, 0)]:
-        return None
-    if node.op.filter_dilation != (1, 1, 1):
-        return None
-
-    bias = theano.tensor.zeros_like(kern[:, 0, 0, 0, 0])
-
-    # need to flip the kernel if necessary (conv3D does not flip)
-    if node.op.filter_flip:
-        kern = kern[:, :, ::-1, ::-1, ::-1]
-
-    # conv3D expects shape (batch, row, column, time, channel)
-    img = img.dimshuffle(0, 2, 3, 4, 1)
-    kern = kern.dimshuffle(0, 2, 3, 4, 1)
-
-    rval = conv3D(img, kern, bias, node.op.subsample)
-    copy_stack_trace(node.outputs[0], rval)
-    rval = rval.dimshuffle(0, 4, 1, 2, 3)
-
-    return [rval]
-
-
-@local_optimizer([AbstractConv2d_gradWeights])
-def local_conv2d_gradweight_cpu(node):
-    if (not isinstance(node.op, AbstractConv2d_gradWeights) or
-            node.inputs[0].dtype == 'float16'):
-        return None
-
-    img, topgrad, shape = node.inputs
-
-    if ((not isinstance(img.type, TensorType) or
-         not isinstance(topgrad.type, TensorType))):
-        return None
-    if node.op.border_mode not in ['full', 'valid']:
-        return None
-    if not node.op.filter_flip:
-        # Not tested yet
-        return
-    if node.op.num_groups > 1:
-        return None
-
-    if node.op.border_mode == 'valid' and \
```

Comment:
This will also request that some tests are removed. There is at least the file nnet/tests/test_conv3d.py that need to be removed.